server/src/models/recipePubRequestModel.js

```diff
@@ -3,8 +3,8 @@ const Schema = mongoose.Schema;
 
 const recipePubRequestSchema = new Schema({
   recipeId: String,
-  userEmail: String
-  // TODO: Add the username of who created
+  userEmail: String,
+  usernameCreator:{type: String,default:"Anonymous"}
```

Comment:
This isn't necessary. You are using the usernameCreator from recipeModel it seems? That other TODO comment should prob just be removed as we don't need it for pub requests.